src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -472,34 +471,37 @@ class CDCL(private val solverType: SolverType = SolverType.INCREMENTAL) : Increm
                     updateLemma(lemma, u)
                 } else if (current != v && !analyzeActivity[current]) {
                     analyzeActivity[current] = true
-                    activeVariables++
+                    numberOfActiveVariables++
                 }
             }
             analyzeActivity[v] = false
-            activeVariables--
+            numberOfActiveVariables--
         }
-        analyzeActivity.indexOfFirst { it }.let { v ->
+
+        var newClause = Clause()
```

Comment:
Why `newClause` is `var`? Why it is called `newClause`?
What are you doing in the following lines? Minimize the learnt clause (`lemma`)? Then okay, you probably need another variable for the new clause. However, you could just mutate `lemma`... But if not, then why not just wrap `lemma` into a `Clause` and return it?
Add some comments!